ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java

```diff
@@ -104,28 +103,44 @@ public OrcSplit(Path path, Object fileId, long offset, long length, String[] hos
     this.isOriginal = isOriginal;
     this.hasBase = hasBase;
     this.rootDir = rootDir;
-    this.deltas.addAll(filterDeltasByBucketId(deltas, AcidUtils.parseBucketId(path)));
+    int bucketId = AcidUtils.parseBucketId(path);
+    long minWriteId = !deltas.isEmpty() ?
+            AcidUtils.parseBaseOrDeltaBucketFilename(path, null).getMinimumWriteId() : -1;
+    this.deltas.addAll(
+            deltas.stream()
+            .filter(delta -> isQualifiedDeleteDeltasByWriteIds(delta, minWriteId))
```

Comment:
maybe simply
````
.filter(delta -> delta.getMaxWriteId() >= minWriteId)
````
for better readability, no need to go inside the method to check the logic?

Reply:
I was thinking it is better to have a method with a good self-describing name for more readability, but I agree that now that the method became very small it is better to do as you suggested. I did that and added a comment on top for explanation.